.github/workflows/functional-test-template.yml

```diff
@@ -0,0 +1,87 @@
+name: OpenSearch Dashboards Functional Test Template
+
+on:
+  workflow_call:
+    inputs:
+      endpoint:
+        required: true
+        type: string
+    secrets:
+      osd-user:
+        required: true
+      osd-user-password:
+        required: true
+
+jobs:
+  OSD-Functional-Test-OSD:
+      runs-on: ubuntu-latest
+      steps:
+        - name: Checkout functional test repo
+          uses: actions/checkout@v3
+          with:
+            repository: elastic-analytics/opensearch-dashboards-functional-test
+            ref: playground
+
+        - name: Step 0 - Install dependencies for functional test
+          run: |
+            npm install
+            npm install -g yarn
+
+        - name: Step 1 - Check If OSD is Up
+          run: |
+            sleep 30s
+            curl -XGET -k -u '${{secrets.osd-user}}:${{secrets.osd-user-password}}' '${{inputs.endpoint}}/api/status' > $HOME/osd-response.json
+            echo "OSD_STATUS=$(cat $HOME/osd-response.json | jq -r '.status.overall.state')" >> $GITHUB_ENV
+
+        - name: Step 2 - Run Functional Test For OSD
+          if: ${{ env.OSD_STATUS == 'green'}}
+          run: |
+            yarn cypress run --spec "cypress/integration/playground/dashboards/*.js" --config "baseUrl=${{inputs.endpoint}}" --env "openSearchUrl=${{inputs.endpoint}},SECURITY_ENABLED=true,username=${{secrets.osd-user}},password=${{secrets.osd-user-password}}"
+
+  Plugins:
+    runs-on: ubuntu-latest
+    needs: OSD-Functional-Test-OSD
+    strategy:
+      fail-fast: false
+      matrix: 
+        run: ['OSD-Functional-Test-AD', 'OSD-Functional-Test-Alert', 'OSD-Functional-Test-Notifications', 'OSD-Functional-Test-Reports', 'OSD-Functional-Test-Observability', 'OSD-Functional-Test-IndexMan']
```

Comment:
if you wanted you could do something like this: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/.github/workflows/build_and_test_workflow.yml#L126

and then it's just like a template where it echos the name and it has the spec

Reply:
Thanks. I tried it, but it seems like that include will run first than shared steps, which can be used for build something, our pipeline will run something after shared steps, probably it cannot fit here.